### PR TITLE
[JVM_IR] Run KotlinSteppingTests with the JVM_IR backend.

### PR DESCRIPTION
**Pull request by madsager**
Implement support for supressing failures by backend.

### CODE REVIEW

**Comment by madsager** [comment]
@max-kammerer 

I'll work on the failures next.

**Comment by madsager** [comment]
I just realized that I did this on top of the when stepping change. Therefore, this actually has a couple of failures more if landed as is. We should wait until we agree on and land https://github.com/JetBrains/kotlin/pull/3519 which fixes 5 failing tests in this suite.

File: idea/jvm-debugger/jvm-debugger-test/test/org/jetbrains/kotlin/idea/debugger/test/util/KotlinOutputChecker.kt [hunk 1]
```diff
@@ -75,7 +77,20 @@ internal class KotlinOutputChecker(
...
         } else {
             val originalText = FileUtilRt.loadFile(outFile, CharsetToolkit.UTF8)
-            val expected = StringUtilRt.convertLineSeparators(originalText)
+            val ignoreDirectives = mutableListOf<TargetBackend>()
+            val expected = StringUtilRt.convertLineSeparators(originalText).split("\n").filter {
+                if (it.trim().startsWith("// IGNORE_BACKEND: ")) {
+                    val backends = it.substringAfter(":").split(",")
+                    backends.forEach {
+                        val trimmed = it.trim()
+                        if (trimmed == "JVM_IR") ignoreDirectives.add(TargetBackend.JVM_IR)
```
Comment (max-kammerer) [inline idea/jvm-debugger/jvm-debugger-test/test/org/jetbrains/kotlin/idea/debugger/test/util/KotlinOutputChecker.kt:86]: Maybe `ignoreDirecrives.add(TargetBackend.valueOf(trimmed))`?

File: idea/jvm-debugger/jvm-debugger-test/test/org/jetbrains/kotlin/idea/debugger/test/util/KotlinOutputChecker.kt [hunk 2]
```diff
@@ -75,7 +77,20 @@ internal class KotlinOutputChecker(
             LOG.error("Test file created ${outFile.path}\n**************** Don't forget to put it into VCS! *******************")
         } else {
             val originalText = FileUtilRt.loadFile(outFile, CharsetToolkit.UTF8)
-            val expected = StringUtilRt.convertLineSeparators(originalText)
+            val ignoreDirectives = mutableListOf<TargetBackend>()
+            val expected = StringUtilRt.convertLineSeparators(originalText).split("\n").filter {
+                if (it.trim().startsWith("// IGNORE_BACKEND: ")) {
```
Comment (max-kammerer) [inline idea/jvm-debugger/jvm-debugger-test/test/org/jetbrains/kotlin/idea/debugger/test/util/KotlinOutputChecker.kt:82]: Consider `InTextDirectivesUtils.findListWithPrefixes` usage

Comment (madsager) [reply idea/jvm-debugger/jvm-debugger-test/test/org/jetbrains/kotlin/idea/debugger/test/util/KotlinOutputChecker.kt]: Thank you! I ended up simplifying most of this away by just using the utility function to check if the test is ignored for the current backend.

File: idea/jvm-debugger/jvm-debugger-test/test/org/jetbrains/kotlin/idea/debugger/test/util/KotlinOutputChecker.kt (hunk 1 above)
Comment (madsager) [reply idea/jvm-debugger/jvm-debugger-test/test/org/jetbrains/kotlin/idea/debugger/test/util/KotlinOutputChecker.kt]: Gone in current version.

**Comment by max-kammerer** [comment]
There are 4 failed tests on TC for original pr
```
IrKotlinSteppingTestGenerated$Custom.testAfterDefaultParameterValues2Intforg.jetbrains.kotlin.idea.debugger.test
IrKotlinSteppingTestGenerated$Custom.testSmartStepIntoInterfaceImplorg.jetbrains.kotlin.idea.debugger.test
IrKotlinSteppingTestGenerated$StepIntoOnly.testTraitsorg.jetbrains.kotlin.idea.debugger.test
IrKotlinSteppingTestGenerated$StepOver.testWhenWithoutExpressionorg.jetbrains.kotlin.idea.debugger.test
```

File: idea/jvm-debugger/jvm-debugger-test/test/org/jetbrains/kotlin/idea/debugger/test/util/KotlinOutputChecker.kt (hunk 2 above)
Comment (max-kammerer) [reply idea/jvm-debugger/jvm-debugger-test/test/org/jetbrains/kotlin/idea/debugger/test/util/KotlinOutputChecker.kt]: Cool! I forgot about it

**Comment by max-kammerer** [comment]
There is no problem on TC with updated one

**Comment by max-kammerer** [comment]
Thank you!

**Comment by max-kammerer** [comment]
@madsager It's strange that `IrKotlinSteppingTestGenerated$Custom.testFunctionBreakpointInStdlib` test fails on AS 4.0 and 4.1
```
com.intellij.rt.execution.junit.FileComparisonFailure: Actual data differs from file content: functionBreakpointInStdlib.out expected:<...XT!
ReadWrite.!EXT!
[]Disconnected from th...> but was:<...XT!
ReadWrite.!EXT!
[resuming ReadWrite.kt:108
]Disconnected from th...>
com.intellij.rt.execution.junit.FileComparisonFailure: Actual data differs from file content: functionBreakpointInStdlib.out expected:<...XT!
ReadWrite.!EXT!
[]Disconnected from th...> but was:<...XT!
ReadWrite.!EXT!
[resuming ReadWrite.kt:108
]Disconnected from th...>
  at org.jetbrains.kotlin.test.KotlinTestUtils.assertEqualsToFile(KotlinTestUtils.java:481)
```

**Comment by madsager** [comment]
> @madsager It's strange that `IrKotlinSteppingTestGenerated$Custom.testFunctionBreakpointInStdlib` test fails on AS 4.0 and 4.1

That is indeed odd. :-\ It is the first time I'm seeing these types of differences. Are there instructions on how to reproduce these somewhere?

**Comment by madsager** [comment]
Looks like the behavior is the same for IR and non-IR backends at least on as41. They both have the unexpected output, so at least it does not look like something introduced here.

**Pull request merged**